### PR TITLE
added CPD check plugin and gradle base structure

### PR DESCRIPTION
**Pull request by m91snik**
added CPD check plugin and gradle base structure to demonstrate checkstyle, PMD plugins + new CPD one.

code intentionally contains violations

### CODE REVIEW

File: build.gradle [hunk 1]
```diff
@@ -0,0 +1,124 @@
...
+apply plugin: 'pmd'
+apply plugin: 'cpd'
+
+lombok {
+    version = versions.lombok
+    sha256 = null
+}
+
+sourceCompatibility = 1.8
+targetCompatibility = 1.8
```
Comment (stimashev-aurea) [inline build.gradle]: consider `JavaVersion.VERSION_1_8`

File: build.gradle [hunk 2]
```diff
@@ -0,0 +1,124 @@
...
+}
+
+plugins {
+    id 'io.franzbecker.gradle-lombok' version '1.8' 
+    id 'java'
+    id 'de.aaschmid.cpd' version '1.0'
+}
+
+group = 'com.aurea.staticcodeanalysis'
+version = '0.0.1'
```
Comment (stimashev-aurea) [inline build.gradle]: I'd start from 1.0 as it already looks mature :-)

Comment (m91snik) [inline build.gradle]: not sure that it's mature... it's just a demo, so we can use any reasonable value for initial version

File: build.gradle (hunk 1 above)
Comment (m91snik) [inline build.gradle]: yeah

**Comment by m91snik** [comment]
@stimashev-aurea, about tests - pls clarify how they can simplify future changes for this 'demo' code?

File: src/main/java/com/aurea/staticcodeanalysis/util/UtilApp.java [hunk 3]
```diff
@@ -0,0 +1,69 @@
+package com.aurea.staticcodeanalysis.util;
+
+import com.aurea.staticcodeanalysis.model.Method;
+import com.aurea.staticcodeanalysis.model.ReportGroup;
+import lombok.experimental.UtilityClass;
+import lombok.extern.slf4j.Slf4j;
+import org.apache.commons.codec.digest.DigestUtils;
+
+import java.util.Objects;
```
Comment (onur-aurea1) [inline src/main/java/com/aurea/staticcodeanalysis/util/UtilApp.java]: order of imports?

File: src/main/java/com/aurea/staticcodeanalysis/GroupProcessor.java [hunk 4]
```diff
@@ -0,0 +1,193 @@
+package com.aurea.staticcodeanalysis;
+
+import com.aurea.staticcodeanalysis.model.Duplicate;
+import com.aurea.staticcodeanalysis.model.Method;
+import com.aurea.staticcodeanalysis.model.ReportGroup;
+import com.aurea.staticcodeanalysis.util.PredicatesUtil;
+import com.aurea.staticcodeanalysis.util.UtilApp;
+import lombok.experimental.UtilityClass;
+
+import java.util.*;
```
Comment (onur-aurea1) [inline src/main/java/com/aurea/staticcodeanalysis/GroupProcessor.java]: .* style import

File: build.gradle [hunk 5]
```diff
@@ -0,0 +1,124 @@
...
+
+cpdCheck {
+    reports {
+        text.enabled = false
+        xml.enabled = true
+    }
+    source = sourceSets.main.allJava // only java, groovy and scala classes in 'main' sourceSets
+}
+
+// CPD settings - end
```
Comment (onur-aurea1) [inline build.gradle]: eof new line

**Comment by m91snik** [comment]
@onur-aurea1 code in this PR should not met standards, it's intentionally left with violations to demonstrate how checkstyle, PMD and CPD works.
Moreover - it's expected that each of these checkers should find some violations to demonstrate how it works

**Pull request merged**